components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/main/java/org/wso2/carbon/policy/mgt/core/mgt/impl/PolicyManagerImpl.java

```diff
@@ -249,21 +251,51 @@ public boolean deletePolicy(Policy policy) throws PolicyManagementException {
                 log.warn("Error occurred while roll backing the transaction.");
             }
             String msg = "Error occurred while deleting the policy ("
-                    + policy.getId() + " - " + policy.getPolicyName() + ")";
+                         + policy.getId() + " - " + policy.getPolicyName() + ")";
+            log.error(msg, e);
+            throw new PolicyManagementException(msg, e);
+        } catch (ProfileManagerDAOException e) {
+            try {
+                PolicyManagementDAOFactory.rollbackTransaction();
+            } catch (PolicyManagerDAOException e1) {
+                log.warn("Error occurred while roll backing the transaction.");
+            }
+            String msg = "Error occurred while deleting the profile for policy ("
+                         + policy.getId() + ")";
+            log.error(msg, e);
+            throw new PolicyManagementException(msg, e);
+        } catch (FeatureManagerDAOException e) {
+            try {
+                PolicyManagementDAOFactory.rollbackTransaction();
+            } catch (PolicyManagerDAOException e1) {
+                log.warn("Error occurred while roll backing the transaction.");
+            }
+            String msg = "Error occurred while deleting the profile features for policy ("
+                         + policy.getId() + ")";
             log.error(msg, e);
             throw new PolicyManagementException(msg, e);
         }
-        return bool;
+
     }
 
     @Override
     public boolean deletePolicy(int policyId) throws PolicyManagementException {
-        boolean bool;
+
         try {
             PolicyManagementDAOFactory.beginTransaction();
+            Policy policy = policyDAO.getPolicy(policyId);
             policyDAO.deleteAllPolicyRelatedConfigs(policyId);
-            bool = policyDAO.deletePolicy(policyId);
+            policyDAO.deletePolicy(policyId);
+
+            if (log.isDebugEnabled()) {
+                log.debug("Profile ID: " + policy.getProfileId());
+            }
+
+            featureDAO.deleteFeaturesOfProfile(policy.getProfileId());
+
+            profileDAO.deleteProfile(policy.getProfileId());
             PolicyManagementDAOFactory.commitTransaction();
+            return true;
```

Comment:
This method always returns true. So can't we simply have void or change the logic to return a bool value depending on the status?